content/docs/user-guide/assets/pipeline/asset-dependencies-and-identifiers.md

```diff
@@ -7,16 +7,22 @@ toc: true
 ---
 
 Both source and product assets might have dependencies. Source asset and job dependencies are inputs to the **Asset Pipeline** and asset processing jobs. Product dependencies are outputs from the Asset Pipeline and are used to load and package product assets. Asset identifiers help ensure both types of asset dependencies are met.
+
+## Source dependencies
+
+Source dependencies are a process time concept.
+
+A source dependency triggers **Asset Processor** to rerun the process job whenever the file declared as a source dependency changes. For example, a `.assetinfo` file is a source dependency and triggers a process job on its source asset when it is modified.
 
-## Source and job dependencies
+## Job dependencies
 
-Source and job dependencies are a process time concept.
+Job dependencies are a process time concept.
 
-A source dependency triggers **Asset Processor** to rerun the process job whenever a source asset changes. For example, a `.assetinfo` file is a source dependency and triggers a process job on its source asset when it is modified.
-
-A job dependency requires the product asset of another process job before this process job can be run. The shader and material relationship is an example of a job dependency. The material cannot be processed until the shader product asset is generated because the material requires information about the shader product asset.
+A job dependency is declared against another job. If a job matching that description exists in the job queue, when the job with the dependency is up next to run, it will first make sure all of its job dependencies are run. Also, if a job dependency runs again, it will trigger any jobs with that dependency to run. The shader and material relationship is an example of a job dependency. The material cannot be processed until the shader job is completed because the material requires information from product assets of the shader job. If the source shader asset is changed, causing the shader job to re-run, then the material job with that job dependency will re-run after the shader job finishes.
 
 Required job dependencies cause the process job to fail if the required product asset is not available, however, job dependencies can be marked optional. Optional job dependencies make use of the requested product asset if it exists, but can complete without the requested product asset.
+
+Job dependencies can be declared on a product asset of the associated job. If this is done, then the job that is the dependency will only cause the job with the dependency to re-run if the hash of the contents of the specified product asset change. This is best used when the dependency has multiple product assets but the job with the dependency only references one of those product assets.
```

Comment:
I would suggest: "Job dependencies can optionally specify a set of product assets from the associated job".  Also perhaps update the wording to be plural, they can specify multiple products

Reply:
Fixed